snuba/migrations/groups.py

```diff
@@ -0,0 +1,45 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+from importlib import import_module
+
+from snuba.migrations.migration import Migration
+
+
+class MigrationGroup(Enum):
+    SYSTEM = "system"
+    EVENTS = "events"
+
+
+class GroupLoader(ABC):
+    """
+    Provides the list of migrations associated with an group and a loader that returns
+    the requested migration.
+    """
+
+    @abstractmethod
+    def load_migration(self, migration_id: str) -> Migration:
+        raise NotImplementedError
+
+
+class DirectoryLoader(GroupLoader):
+    """
+    Loads migrations that are defined as files of a directory. The file name
+    represents the migration ID.
+    """
+
+    def __init__(self, module: str) -> None:
+        self.__module = module
+
+    def load_migration(self, migration_id: str) -> Migration:
```

Comment:
If the runner gets an innstance of group loader and the group loader has a method to return an individual migration given the id, who gives the list of valid ids to the Runner ?

Reply:
Group loader should have a function to return all the migration IDs for that group. This function will likely be changed quite a lot when we add the management of dependencies and detecting what needs to be run.

Reply:
ok